tools/spatialDraw/DrawingManager.js

```diff
@@ -569,67 +630,273 @@ DrawingManager.Tool.Line = class extends DrawingManager.Tool {
+        }
+        console.log(`Drawing ${drawing.iconName} from serialized.`);
+        console.log(drawing);
+        const selectedIcon = this.icons[drawing.iconName];
+        const obj = selectedIcon.scene.clone();
+        obj.iconName = selectedIcon.iconName;
+        obj.traverse(child => {
+            if (child.material) {
+                child.material = new THREE.MeshLambertMaterial({color: drawing.color});
+            }
+        });
+        parent.add(obj);
+        obj.scale.multiplyScalar(drawing.size * 10);
+        obj.position.copy(drawing.position);
+
+        drawing.normalVector = new THREE.Vector3(drawing.normalVector.x, drawing.normalVector.y, drawing.normalVector.z);
+        // Note: I do not know why (mathematically) the normal needs to be negated here. If not negated, it points the wrong direction.
+        const transformedNormal = drawing.normalVector.clone().negate().applyMatrix4(this.drawingManager.scene.matrixWorld.clone().invert().transpose());
+        obj.lookAt(obj.getWorldPosition().add(transformedNormal));
+        obj.normalVector = transformedNormal;
+
+        obj.animationMixer = new THREE.AnimationMixer(obj);
+        obj.lastRender = 0;
+        this.drawingManager.addCallback('render', _now => {
+            const deltaSeconds = (_now - obj.lastRender) / 1000;
+            obj.lastRender = _now;
+            obj.animationMixer.update(deltaSeconds);
+        });
+        selectedIcon.animations.forEach(animation => {
+            obj.animationMixer.clipAction(animation).play();
+        });
+
+        obj.drawingId = drawing.drawingId;
+        obj.serialized = drawing;
+    }
+
+    /**
+     * Sets the drawing size.
+     * @param {number} size - The size.
+     */
+    setSize(size) {
+        super.setSize(size);
+    }
+
+    /**
+     * Sets the drawing color.
+     * @param {string} color - The color.
+     */
+    setColor(color) {
+        super.setColor(color);
     }
 };
 
-DrawingManager.Cursor.Projection = class extends DrawingManager.Cursor {
+DrawingManager.Cursor.Offset = class extends DrawingManager.Cursor {
     /**
-     * Creates a Projection Cursor.
+     * Creates an Offset Cursor.
      */
     constructor() {
         super();
+        this.offset = 500;
         this.position = new THREE.Vector3(0, 0, 0);
     }
 
     /**
-     * Updates the cursor position.
+     * Updates the cursor position and normal.
      * @param {THREE.Scene} scene - The scene to calculate the position in.
      * @param {THREE.Camera} camera - The camera used for calculating the cursor position.
      * @param {Object} pointerEvent - The triggering pointer event.
      */
     updatePosition(scene, camera, pointerEvent) {
-        const offset = pointerEvent.projectedZ;
-        if (!offset) {
-            return;
-        }
-        this.position = super.screenProject(pointerEvent, offset, camera, scene);
+        this.position = super.screenProject(pointerEvent, this.offset, camera, scene);
```

Comment:
Can use this.screenProject to be more friendly to any future implementation of screenProject